cmd/comfy/price_cache.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"errors"
+	"log"
+	"strconv"
+	"time"
+)
+
+type PriceCache struct {
+	prices        map[string]*HourlyPrice
+	fetchCallback FetchPrices
+}
+
+type FetchPrices = func() []HourlyPrice
+
+func initPriceCache(fetchCallback FetchPrices) *PriceCache {
+	var priceCache = &PriceCache{prices: nil, fetchCallback: fetchCallback}
+
+	priceCache.refreshPrices(priceCache.fetchCallback())
+
+	return priceCache
+}
+
+func (priceCache *PriceCache) refreshPrices(pricesToday []HourlyPrice) {
+	log.Printf("Refreshing prices!")
+	priceCache.prices = make(map[string]*HourlyPrice, 48)

```

Comment:
maybe we can use `len(pricesToday)` here, so that we always allocate just enough for all values received from api?